### PR TITLE
Add support for ALB

### PR DESCRIPTION
**Pull request by AndreKurait**
### Description
Adds configuration option to support deploying with ALB

### Issues Resolved
N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: lib/infra/infra-stack.ts [hunk 1]
```diff
@@ -1013,4 +1044,75 @@ export class InfraStack extends Stack {
...
+      default:
+        throw new Error('Unsupported load balancer type.');
+    }
+  }
+
+  /**
+   * Adds targets to the given listener.
+   * Works for both Application Load Balancers and Network Load Balancers.
+   */
+  private static addTargets(listener: BaseListener, elbType: LoadBalancerType, id: string, port: number, target: AutoScalingGroup | InstanceTarget,
```
Comment (rishabh6788) [inline lib/infra/infra-stack.ts:1093]: the method name confuses with `addTargets` method of [ApplicationListener](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_elasticloadbalancingv2.ApplicationListener.html). Will it better to rename it avoid confusion?

File: test/opensearch-cluster-cdk.test.ts [hunk 2]
```diff
@@ -1070,3 +1070,50 @@ test('Ensure target group protocol is always TCP', () => {
...
   });
 });
+
+
+describe.each([
+  { loadBalancerType: 'alb', securityDisabled: false, expectedType: 'application', expectedProtocol: 'HTTPS' },
+  { loadBalancerType: 'alb', securityDisabled: true, expectedType: 'application', expectedProtocol: 'HTTP' },
+  { loadBalancerType: 'nlb', securityDisabled: false, expectedType: 'network', expectedProtocol: 'TLS' },
+  { loadBalancerType: 'nlb', securityDisabled: true, expectedType: 'network', expectedProtocol: 'TCP' },
+])('Test $loadBalancerType creation with securityDisabled=$securityDisabled', ({ loadBalancerType, securityDisabled, expectedType, expectedProtocol }) => {
```
Comment (rishabh6788) [inline test/opensearch-cluster-cdk.test.ts:1080]: Can you add one case for throwing exception when unexpected input is provided?

Comment (AndreKurait) [reply test/opensearch-cluster-cdk.test.ts:1080]: Added

**Review by rishabh6788 — APPROVED** [review]
LGTM!
Thank you for your contribution.

**Pull request merged**